File: go/RemoveDuplicateFromSortedLL/RemoveDuplicateFromSortedLL.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

// Time -> 0(N) Space -> 0(1)
func deleteDuplicates(head *ListNode) *ListNode {
	stillHead := head
	for head != nil {
		currHead := head
		head = head.Next
		for head != nil && currHead.Val == head.Val {
			head = head.Next
		}
		currHead.Next = head
	}
	return stillHead
}

File: python/removeDuplicateFromSortedLL.py
class LinkedList:
    def __init__(self, val=0, next=None) -> None:
        self.val = val
        self.next = next


# Time -> 0(N) Space -> 0(1)
def removeDuplicates(head) -> LinkedList:
    stillHead = head
    while head is not None:
        currHead = head
        head = head.next
        while head is not None and currHead.val == head.val:
            head = head.next
        currHead.next = head
    return stillHead
